cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -243,11 +245,19 @@ func getClusterSharding(kubeClient *kubernetes.Clientset, settingsMgr *settings.
 	// StatefulSet mode and Deployment mode uses different default values for shard number.
 	defaultShardNumberValue := 0
 	applicationControllerName := env.StringFromEnv(common.EnvAppControllerName, common.DefaultApplicationControllerName)
```

Comment:
Can we move this line inside the if block on line 255? This `applicationControllerName` is used only for app controller deployment logic.

Reply:
Fixed